pkg/workflows/dependency_graph.go

```diff
@@ -15,29 +15,34 @@ const (
 	KeywordTrigger = "trigger"
 )
 
+type StepInputs struct {
+	OutputRef string
+	Mapping   map[string]any
+}
+
 // StepDefinition is the parsed representation of a step in a workflow.
 //
 // Within the workflow spec, they are called "Capability Properties".
 type StepDefinition struct {
-	ID     string         `json:"id" jsonschema:"required"`
```

Comment:
Why all the json tags are not needed any more?

Reply:
These weren't actually used for anything aside from tests, so I removed it. I found it confusing for example to see the jsonschema tags since that implied we enforced this somewhere, but we don't.

Reply:
I applied the same logic to the json tags: since they aren't used, let's remove them so as not to imply that this struct is serialized.